sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/Utils.java

```diff
@@ -93,18 +93,48 @@ private static ObjectMapper createAndInitializeObjectMapper(boolean allowDuplica
         }
         objectMapper.configure(DeserializationFeature.ACCEPT_FLOAT_AS_INT, false);
 
-
-        // We will not register after burner for java 16+, due to its breaking changes
-        // https://github.com/Azure/azure-sdk-for-java/issues/23005
-        if (JAVA_VERSION != -1 && JAVA_VERSION < 16) {
-            objectMapper.registerModule(new AfterburnerModule());
-        }
+        tryToLoadJacksonPerformanceLibrary(objectMapper);
 
         objectMapper.registerModule(new JavaTimeModule());
 
         return objectMapper;
     }
 
+    private static void tryToLoadJacksonPerformanceLibrary(ObjectMapper objectMapper) {
+        // Afterburner and Blackbird are libraries that increase the performance of marshaling json to objects
+        boolean loaded = false;
+        if (JAVA_VERSION != -1) {
+            if (JAVA_VERSION >= 11) {
+                // Blackbird is preferred and only works with java 11+
+                // https://github.com/FasterXML/jackson-modules-base/tree/2.18/blackbird
+                loaded = loadModuleIfFound("com.fasterxml.jackson.module.blackbird.BlackbirdModule", objectMapper);
```

Comment:
will shading impact the path here?

Reply:
If it was shaded, then the original class would be `<shade>.com.fasterxml.jackson.module.afterburner.AfterburnerModule`, but it's not referred to that in the code today.

Reply:
the code currently uses `new AfterburnerModule()` directly so it will still work for shading. But when we changing to use class name, I think we need to a way to refer to it properly even when shading happens...

Reply:
shading will also update all strings - not just in import statement and package statement - so it will automatically shae Blackbird as well.

Reply:
Gotcha, missed the fact that simple string will also get updated. Thanks

Reply:
I don't think afterburner is being shaded at all... If it was shaded, I wouldn't need to add it to the classpath for the azure sdk to function.  Am I missing something?

Reply:
Sorry, misunderstanding. azure-cosmos is not shading it. But the Spark connector azure-cosmos-spark-* is shading dependencies (azure-core, azure-cosmos, netty, reactor, jackson etc.). And Annie's concerns were related to validating whether this would cause issues int he Spark connector or the Kafka connector (also using shading similar to spark)